client/src/util/fetchFromAPI.test.js

```diff
@@ -1,10 +1,11 @@
-import { it, expect } from "vitest";
+import { expect, test } from "@jest/globals";
 import { queryAPI } from "./fetchFromAPI";
 
-//Mocking fetch calls
 //If the fetch call results in a 200 response, the onSuccess callback is fired
-it("should return a 200 status code", () => {
-  expect(queryAPI).toBe(200);
+test("the data is successfully fetched from the API", () => {
+  return queryAPI().then((data) => {
+    expect(data).toBe("status code 200");
+  });
```

Comment:
I see you're using a promise here (i.e. `queryAPI().then()`).  Remember that the design of the `queryAPI` function makes use of success/error callbacks, not promises.

Now that I see the tests for this I'm rethinking the callback strategy. They are much more complicated to test than promises.  What are your thoughts on redesigning the `queryAPI` function to use promises instead of callbacks to simplify the testing?

Reply:
<img width="453" alt="Screen Shot 2023-01-23 at 1 21 21 PM" src="https://user-images.githubusercontent.com/34128735/214119431-13909ea6-9106-4298-b2eb-3655e2116cda.png">

Pushed new changes - How does this new implementation look? I ran the npm test to give me a visual of what the test is returning